File: src/main/java/hilfling/backend/hilfling/controllers/PhotographyRequestController.java
package hilfling.backend.hilfling.controllers;

import hilfling.backend.hilfling.model.PhotographyRequest;
import hilfling.backend.hilfling.service.PhotographyRequestService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/v1/photography_requests")
public class PhotographyRequestController extends GenericBaseControllerImplementation<PhotographyRequest> {
    @Autowired
    private PhotographyRequestService service;
    public PhotographyRequestService getService() {
        return service;
    }
}

File: src/main/kotlin/hilfling/backend/hilfling/controller/PhotographyRequestController.kt
package hilfling.backend.hilfling.controller

import hilfling.backend.hilfling.model.Album
import hilfling.backend.hilfling.model.PhotographyRequest
import hilfling.backend.hilfling.repository.AlbumRepository
import hilfling.backend.hilfling.repository.PhotographyRequestRepository
import me.liuwj.ktorm.database.Database
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/photography_requests")
class PhotographyRequestController {
    val repository = PhotographyRequestRepository()

    @GetMapping("/{id}")
    fun getById(@PathVariable("id") id: Long) : PhotographyRequest? {
        return repository.findById(id)
    }

    @GetMapping
    fun getAll() : List<PhotographyRequest> {
        return repository.findAll()
    }

    @PostMapping
    fun create(
            @RequestParam("photographyRequest") photographyRequest: PhotographyRequest
    ): PhotographyRequest {
        return repository.create(photographyRequest)
    }
}